### PR TITLE
New monorepo structure

### PR DESCRIPTION
**Pull request by simonswiss**
This PR removes the `bolt` dependency, and brings a few useful tools to work with a monorepo:

- [yarn workspaces](https://yarnpkg.com/lang/en/docs/workspaces/) - setup multiple packages in such a way that you only need to run yarn install once to install all of them in a single pass
- [changesets](https://github.com/atlassian/changesets) - A way to manage your versioning and changelogs with a focus on multi-package repositories
- [preconstruct](https://preconstruct.tools/) - A smart build tool for libraries
- [manypkg](https://github.com/mitchellhamilton/manypkg) - An umbrella for your multi-package repo

The PR brings a few changes:

## Components are built with preconstruct

No more build scripts for the packages in `components` - preconstruct takes care of bundling them. Run `yarn` from the rool, and `preconstruct dev` will be run as a `postinstall` script.

## Single script to build the brands

A new file, `./helpers/buildBrands.js`, is responsible for importing the web transformer, and run it against each brand in the `bands` folder.

### CODE REVIEW

File: helpers/transformers/web.js [hunk 1]
```diff
@@ -38,8 +36,9 @@ function convertFonts(fonts) {
  * Transform the tokens into a web specific object that can be used in emotion
  *
  * @param  {string} BRAND - The brand string to find the right brand folder
+ * @param  {string} dest - The destination path where the file should be written
  */
-function build(BRAND) {
+function build(BRAND, dest = '.') {
```
Comment (simonswiss) [inline helpers/transformers/web.js]: Passed a default value of current directory here to preserve the existing behaviour if no second argument is passed - not sure if this is the best way to approach it.

File: package.json [hunk 2]
```diff
@@ -20,29 +20,27 @@
...
+		"start": "/bin/sh -c 'cd components/${1:-$0} && yarn start'",
 		"docs": "docz dev",
-		"docs:build": "docz build && bolt docs:redirects",
+		"docs:build": "docz build && yarn docs:redirects",
 		"docs:redirects": "cp helpers/public/* docs/",
-		"nuke": "rm -rf node_modules && bolt ws exec -- rm -rf node_modules",
-		"build": "bolt ws exec -- rm -rf dist && bolt ws exec -- bolt build",
-		"fresh": "bolt nuke && bolt install",
+		"nuke": "find ./components -type d -name 'node_modules' -exec rm -rf {} + && find ./brands -type d -name 'node_modules' -exec rm -rf {} +",
+		"build": "node ./helpers/buildBrands.js",
```
Comment (mitchellhamilton) [inline package.json]:
```suggestion
		"build": "preconstruct build && node ./helpers/buildBrands.js",
```

File: helpers/transformers/web.js [hunk 3]
```diff
@@ -75,7 +74,7 @@ function build(BRAND) {
 	};
 
 	try {
-		fs.mkdirSync('dist/');
+		fs.mkdirSync(`${dest}/dist/`);
```
Comment (mitchellhamilton) [inline helpers/transformers/web.js]:
```suggestion
		fs.mkdirSync(path.join(dest, 'dist'));
```

File: README.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ The design system for Westpac GEL
 | `bolt dev [package-name]`   | run the examples of the specified component       |
 | `bolt new [package-name]`   | create a specified empty component                |
 | `bolt start [package-name]` | start the example server of a component           |
-| `bolt test`                 | runs test                                         |
+| `yarn test`                 | runs test                                         |
```
Comment (mitchellhamilton) [inline README.md]: All the things should use yarn

File: brands/BOM/package.json [hunk 5]
```diff
@@ -18,8 +18,15 @@
 	],
 	"license": "GPL-3.0+",
 	"scripts": {
-		"prepublish": "node index.js",
-		"build": "node index.js"
+		"prepublish": "yarn build",
+		"build": "rm -rf dist && node index.js"
```
Comment (mitchellhamilton) [inline brands/BOM/package.json]: These things shouldn't exist

File: brands/BSA/package.json [hunk 6]
```diff
@@ -18,8 +18,15 @@
...
 	},
-	"main": "dist/index.js"
+	"main": "dist/index.js",
+	"dependencies": {},
+	"peerDependencies": {
+		"color": "^3.1.1"
+	},
+	"devDependencies": {
+		"color": "^3.1.2"
+	}
```
Comment (mitchellhamilton) [inline brands/BSA/package.json]: These should probably go

File: helpers/buildBrands.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+	workspaces
+		.filter(workspace => workspace.dir.includes('brand'))
+		.map(workspace => {
+			const { name, dir } = workspace;
+			const BRAND = name.replace('@westpac/', '').toUpperCase();
+
+			// Remove the dist folder
+			fs.removeSync(`${dir}/dist`);
+
+			build(BRAND, dir);
```
Comment (mitchellhamilton) [inline helpers/buildBrands.js]: Maybe make all the things async or maybe it's fine, idk

File: helpers/cli.js [hunk 8]
```diff
@@ -127,7 +127,7 @@ const main = async () => {
 		console.log(chalk.gray(`\nCreating ${component.key} component...\n`));
 		createComponent(component);
 		console.log(chalk.gray('[3/3] Installing dependencies...\n'));
-		shell(`bolt`);
+		shell(`yarn`);
```
Comment (mitchellhamilton) [inline helpers/cli.js]:
```suggestion
		shell(`yarn`, { stdio: 'inherit' });
```

File: helpers/transformers/web.js (hunk 1 above)
Comment (mitchellhamilton) [reply helpers/transformers/web.js]: dest isn't necessary since if you know the brand, you also know the directory.

File: package.json (hunk 2 above)
Comment (simonswiss) [reply package.json]: `preconstruct build` exposes a few issues with the repo - I have fixed a couple but will remove that for now, noting that this error needs to be fixed:

```🎁 error @westpac/alert { TypeError: Cannot read property 'id' of null
🎁 error @westpac/alert     at Object.resolveId (/Users/simonswiss/Sites/thinkmill/westpac/GEL/node_modules/preconstruct/cli/dist/preconstruct.cjs.dev.js:1055:20)
🎁 error @westpac/alert     at <anonymous>
🎁 error @westpac/alert   code: 'PLUGIN_ERROR',
🎁 error @westpac/alert   plugin: 'flow-and-prod-dev-entry',
🎁 error @westpac/alert   hook: 'resolveId' }```

**Pull request merged**